From: Jakub Jirutka <jakub@jirutka.cz>
Date: Tue, 12 Jul 2022 16:49:54 +0200
Subject: [PATCH] Don't use old SSL names (OpenSSL macros)

These macros are defined by OpenSSL and are available in WolfSSL only
when built with OpenSSL compatibility layer.

--- a/lib/riemann/client/tls-wolfssl.c
+++ b/lib/riemann/client/tls-wolfssl.c
@@ -162,42 +162,42 @@
 
   conn->tls.ctx = wolfSSL_CTX_new(wolfTLSv1_2_client_method());
   if (conn->tls.ctx == NULL) {
-    e = SSL_FAILURE;
+    e = WOLFSSL_FAILURE;
     goto end;
   }
 
   e = wolfSSL_CTX_load_verify_locations(conn->tls.ctx,
                                         conn->tls.options.cafn, 0);
-  if (e != SSL_SUCCESS) {
+  if (e != WOLFSSL_SUCCESS) {
     goto end;
   }
 
   e = wolfSSL_CTX_use_certificate_file(conn->tls.ctx,
                                        conn->tls.options.certfn,
-                                       SSL_FILETYPE_PEM);
-  if (e != SSL_SUCCESS) {
+                                       WOLFSSL_FILETYPE_PEM);
+  if (e != WOLFSSL_SUCCESS) {
     goto end;
   }
 
   e = wolfSSL_CTX_use_PrivateKey_file(conn->tls.ctx,
                                       conn->tls.options.keyfn,
-                                      SSL_FILETYPE_PEM);
-  if (e != SSL_SUCCESS) {
+                                      WOLFSSL_FILETYPE_PEM);
+  if (e != WOLFSSL_SUCCESS) {
     goto end;
   }
 
   if (conn->tls.options.priorities)
     {
       if (wolfSSL_CTX_set_cipher_list(conn->tls.ctx,
-                                      conn->tls.options.priorities) != SSL_SUCCESS) {
-        e = SSL_FAILURE;
+                                      conn->tls.options.priorities) != WOLFSSL_SUCCESS) {
+        e = WOLFSSL_FAILURE;
         goto end;
       }
     }
 
   conn->tls.ssl = wolfSSL_new(conn->tls.ctx);
   if (conn->tls.ssl == NULL) {
-    e = SSL_FAILURE;
+    e = WOLFSSL_FAILURE;
     goto end;
   }
 
@@ -212,7 +212,7 @@
 
   e = _riemann_client_set_timeout_socket (client, &timeout);
   if (e != 0) {
-    e = SSL_FAILURE;
+    e = WOLFSSL_FAILURE;
     goto end;
   }
 
@@ -222,7 +222,7 @@
 
  end:
 
-  if (e != SSL_SUCCESS) {
+  if (e != WOLFSSL_SUCCESS) {
     client->disconnect (client);
     return -EPROTO;
   }
